### PR TITLE
Integrating composite triggers with the `DeploymentTrigger` YAML representation

### PR DESCRIPTION
**Pull request by chrisguidry**
It wasn't apparent to me the first time that just backporting the `Trigger` hierarchy and `TriggerTypes` wasn't enough to get them to be available in deployments.  This adds them to the `DeploymentTrigger` schema so that folks can define them in their `prefect.yaml` files and get full support of the nested triggers.

### CODE REVIEW

**Review by serinamarie — COMMENTED** [review]
Do you think it'd be a good idea to add some integration tests like in our
`test_deployments.py`
 - `TestDeploymentBasicInterface`
 - `TestDeploymentApply`
 
and like tests for malformed triggers that are missing required fields?

**Comment by chrisguidry** [comment]
That's a great call, @serinamarie, I'll try to get some of those in today

**Comment by chrisguidry** [comment]
It turns out that we do have a set of integration tests over in `tests/cli/test_deploy.py` to make sure that we can parse these into automations when creating deployments, so that's good.

I'm still not satisfied with how this works, so I'm going to do some more work on it to reflect the real triggers class hierarchy

**Review by jakekaplan — APPROVED** [review]

File: tests/events/client/test_deployment_trigger_schema.py [hunk 1]
```diff
@@ -207,6 +207,93 @@ def test_compound_deployment_trigger_as_automation():
     )
 
 
+def test_deeply_nested_compound_deployment_trigger_as_automation():
```
Comment (serinamarie) [inline tests/events/client/test_deployment_trigger_schema.py:210]: ❤️

**Review by serinamarie — APPROVED** [review]

**Pull request merged**